CoreScriptsRoot/Modules/VR/NotificationHub.lua

```diff
@@ -0,0 +1,730 @@
+								notification.detailsFrame.Size = UDim2.new(1, -DETAILS_PADDING * 2, fraction, -DETAILS_PADDING * 2)
+
+								detailY = detailY + fraction
+							else
+								notification.detailsFrame.Visible = false
+							end
+						end
+					end
+				end
+			end
+		end
+	end
+end
+
+local NotificationGroup = {} 
+do
+	local NotificationGroup_mt = { __index = NotificationGroup }
+	function NotificationGroup.new(key, title, order) 
+		local self = setmetatable({}, NotificationGroup_mt)
+
+		self.key = key
+		self.title = title
+		self.order = order
+		self.notifications = {}
+		self.notificationsDirty = false
+		self.frame = Util:Create "Frame" {
+			Parent = notificationsWindow.content,
+			BackgroundTransparency = 1
+		}
+		self.detailsFrame = Util:Create "Frame" {
+			Parent = nil,
+			BackgroundTransparency = 1,
+
+			Position = UDim2.new(0, 0, 0, 0),
+			Size = UDim2.new(1, 0, 1, 0)
+		}
+
+		notificationsGroups[key] = self
+		table.insert(notificationsGroupsList, self)
+
+		return self
+	end
+
+	function NotificationGroup:Deactivate()
+		self.detailsFrame.Parent = nil
+		for i, v in pairs(self.detailsFrame:GetChildren()) do
+			detailsPanel:RemoveSubpanel(v)
+		end
+	end
+
+	function NotificationGroup:SwitchTo()
+		detailsWindow:SetTitle(self.title)
+		for i, v in pairs(notificationsGroups) do
+			if v ~= self then
+				v:Deactivate()
+			end
+		end
+		self.detailsFrame.Parent = detailsWindow.content
+
+		activeGroup = self
+		self.notificationsDirty = true
+	end
+
+	function NotificationGroup:BringNotificationToFront(notification)
+		if activeGroup ~= self then
+			self:SwitchTo()
+		end
+
+		for i, v in ipairs(self.notifications) do
+			if v == notification then
+				if i == #self.notifications then
+					layoutNotificationsGroups()
+					return --already on top, no point
+				end
+				--take it out
+				table.remove(self.notifications, i)
+				break
+			end
+		end
+		--put it back on top
+		table.insert(self.notifications, notification)
+		self.notificationsDirty = true
+		layoutNotificationsGroups()
+	end
+
+	function NotificationGroup:RemoveNotification(notification)
+		for i, v in ipairs(self.notifications) do
+			if v == notification then
+				table.remove(self.notifications, i)
+				notificationsPanel:RemoveSubpanel(notification.frame)
+				detailsPanel:RemoveSubpanel(notification.detailsFrame)
+				notification.detailsFrame:Destroy()
+				notification.frame:Destroy()
+
+				self.notificationsDirty = true
+				layoutNotificationsGroups()
+				return
+			end
+		end
+	end
+
+	function NotificationGroup:GetTopNotification()
+		local numNotifications = #self.notifications
+		if numNotifications <= 0 then
+			return nil
+		end
+		return self.notifications[numNotifications]
+	end
+end
+
+NotificationGroup.new("Friends", 		"Friends",	1)
+NotificationGroup.new("BadgeAwards", 	"Badges", 	2)
+NotificationGroup.new("PlayerPoints",	"Points", 	3)
+NotificationGroup.new("Developer", 		"Other", 	4)
```

Comment:
The spacing on these lines are super funny, I think it is a problem with the tabs? just for formatting lets not indent the args.